asd-aituber/apps/web/__tests__/invalid-state-error-handling.test.ts

```
@@ -0,0 +1,945 @@
+      ;(global as any).SpeechRecognition = originalSpeechRecognition
+      ;(global as any).webkitSpeechRecognition = originalSpeechRecognition
+    }
+  })
+
+  describe('tdd-2.9: Verify InvalidStateError handling when recognition already running', () => {
+    it('should detect and handle InvalidStateError when recognition is already active', () => {
+      class SpeechRecognitionStateManager {
+        private recognition: any
+        private isRecognitionActive = false
+        private pendingStart = false
+        private stateChangeListeners: Function[] = []
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)